### ISSUE TITLE
Alienfan Not working at insider Windows 11 release

### ISSUE THREAD

**Issue by Hellohi3654** [T-Troll/alienfx-tools#114]
**Describe the bug**
When trying to open alienfan-gui.exe it says can't install driver, check system configuration!
I am using the latest version as of writing this 
I started happening after I updated to the latest windows insider dev build.
I have also tried olders version and they show the same error

**To Reproduce**
1. Download latest windows 11 insider dev build and try and run alienfan-gui.exe
2. See pop-up saying can't install driver, check system configuration!

**Screenshots**
![image](https://user-images.githubusercontent.com/57390764/154785294-e51bc0ab-71d3-4b2b-b18c-a1c1f1712324.png)

**Comment by T-Troll** [comment]
Uff... Quite bad news.

This is mean as it means - i use driver loading hack to load unsigned driver, and seems like this new build block my hack way.

I use 3rd-party library for this, so let me check about it updates.

**Comment by Hellohi3654** [comment]
would it be possible for me to brute force and replace the driver manually?
Just as a quick fix for now?

**Comment by Hellohi3654** [comment]
couldn't it also be that something has changed meaning it doesn't recognize my laptop anymore
As it says supported hardware not detected

**Comment by Hellohi3654** [comment]
hwinfo is still able to control my fans but they use a different method I think

**Comment by T-Troll** [comment]
It's not a hardware issue, but a driver loading issue.
And no, it's not possible to load it other way (i remove it because it can provide random BSODs).

What you can TRY to do:
- Disable secure boot (but i don't sure W11 allow it)
- Switch system into test mode
- Roll back to previous release
...
- Wait until updated driver loader for this release will be avaliable

**Comment by Hellohi3654** [comment]
disabling secure boot didn't help
what driver what I roll back to the previous release?
or do I have to roll back a windows update?
Thanks

**Comment by Hellohi3654** [comment]
would I roll back the intel driver?
Don't really know why there even is one
I have AMD and Nvidia
![image](https://user-images.githubusercontent.com/57390764/154932600-6f3ccf1a-72da-44eb-a6db-89308a300f44.png)

**Comment by T-Troll** [comment]
Driver (hwacc.sys) was not changed for the long time.
So you should roll back release... Or wait until loader update will be published.

**Comment by Hellohi3654** [comment]
hwacc.sys ok

**Comment by T-Troll** [comment]
But now it can't be loaded this way. And i will not pay M$ $150 per year to sign it.

**Comment by Hellohi3654** [comment]
that's fair

**Comment by Hellohi3654** [comment]
where is hwacc.sys located?

**Comment by Hellohi3654** [comment]
hmm
![image](https://user-images.githubusercontent.com/57390764/154934365-8823f990-d5df-4921-b978-0a80a2161043.png)

**Comment by Hellohi3654** [comment]
![image](https://user-images.githubusercontent.com/57390764/154934564-6391b414-c4af-4907-90b3-0e685bec5ef4.png)

**Comment by Hellohi3654** [comment]
am I looking in the right spot?

**Comment by T-Troll** [comment]
The issue is not the driver itself, but the fact i use hack method to load it (old one, it works from XP until W11) - because W10+ doesn't allow loading unsigned drivers without serious effort from the user (from W10).

hwacc.sys is my own driver used to access ACPI BIOS calls, it's located into app folder.

**Comment by Hellohi3654** [comment]
oh ok
well I'm willing to allow unsigned driver loading is there a tutorial I can follow

**Comment by T-Troll** [comment]
It doesn't help. Trying to load it without Test Mode provide BSOD in W11, so i disable this way.

Thank you to reveal me this point, but for now you can only roll back insider release to previous one.
... or, like me, stay at W10.

**Comment by Hellohi3654** [comment]
I was going to stay on win 10 because there were issues with AMD CPU's but then all of my Microsoft store games broke and the only way I could fix them was to update to win 11 unfortunately.

Also would it be possible for you to be able to control the fans the same way hwinfo does since it still seems to work fine

**Comment by Hellohi3654** [comment]
also how do I disable alienfan from starting on boot manually since I can't open it anymore

**Comment by T-Troll** [comment]
> Also would it be possible for you to be able to control the fans the same way hwinfo does since it still seems to work fine
No, it's a dangerous way, as well as many gears not compatible with it at all.

> also how do I disable alienfan from starting on boot manually since I can't open it anymore
Open start menu, start to type "task scheduler", open it, remove `AlienFan-GUI` entry. I'll add automatic remove at next release.

**Comment by T-Troll** [comment]
> I was going to stay on win 10 because there were issues with AMD CPU's but then all of my Microsoft store games broke and the only way I could fix them was to update to win 11 unfortunately.

I broke it from time to time as well, it's two shell commands return it back.

**Comment by T-Troll** [comment]
Ok... I'll keep this task opened, and drop a message here, then i have updates.

**Comment by Hellohi3654** [comment]
> > I was going to stay on win 10 because there were issues with AMD CPU's but then all of my Microsoft store games broke and the only way I could fix them was to update to win 11 unfortunately.
> 
> I broke it from time to time as well, it's two shell commands return it back.

Yeah I tried to use those shell commands but everything I tried didn't fix it
I ended up finding out that there was a cumulative update that was meant to fix my issue so I had to get win 11 to get it.

**Comment by Hellohi3654** [comment]
> Ok... I'll keep this task opened, and drop a message here, then i have updates.

Ok thanks

**Comment by Hellohi3654** [comment]
> > Also would it be possible for you to be able to control the fans the same way hwinfo does since it still seems to work fine
> 
> No, it's a dangerous way, as well as many gears not compatible with it at all.
> 
> > also how do I disable alienfan from starting on boot manually since I can't open it anymore
> 
> Open start menu, start to type "task scheduler", open it, remove `AlienFan-GUI` entry. I'll add automatic remove at next release.

Thanks

**Comment by T-Troll** [comment]
Ok, you can try to find a way to load driver yourself.
I use [this](https://github.com/hfiref0x/KDU) library to load my driver.

So you can try to load utility from there, then try to load my driver using it with different parameters (provider/shellcode).
In case you can load driver successfully (you can run alienfan-gui/cli for check), please share provider/shellcode IDs for me, i can alter app settings.

If you can't load it with any combination - we have no choice but wait until guys update it.

WARNING: using incorrect provider/shell can provide BSOD. Close all other apps, use with care!

**Comment by Hellohi3654** [comment]
I experimented with kdu and found that this command loads your driver just fine
no BSODs and I have tested it and it seems to be 100% working
"kdu.exe -prv 1 -scv 3 -drvn HwAcc -map HwAcc.sys"

**Comment by T-Troll** [comment]
Great news!
Ok, i'll check it at W10 and then publish update.

You can load driver by this command meanwhile before `AlienFan-GUI` start.

**Comment by Hellohi3654** [comment]
Yeah I have setup a bat file that runs on startup that loads the driver
then opens alienfan-gui using task scheduler

**Comment by T-Troll** [comment]
Ok, let's try the easy way first.
This is [updated loader DLL](https://github.com/T-Troll/alienfx-tools/files/8150233/kdl.zip).
Please overwrite kdl.dll into app folder by it and try (NB: don't forget to disable your current loading script and reboot before it!).

If it not works i need more deep library update, it took some days...

**Comment by Hellohi3654** [comment]
your new version 5.5.4 seems to work fine

**Comment by Hellohi3654** [comment]
Have you tried using version 5.5.5.1
5.5.7 seems to break my fans 
Haven't tried 5.5.6

**Comment by T-Troll** [comment]
@VinyLimaZ What gear do you have and what exactly happened?

@Hellohi3654 Oh... Can you open an issue about it at alienfx-tools with more details? I have no changes in fan control (except some for G5, but it should not break anything) after 5.5.5.1, so let's check and fix.

**Comment by VinyLimaZ** [comment]
> 

@Hellohi3654 No I tried the same version you used (5.5.4) but didn't work :/ I'll try that 5.5.5.1 later! TY

@T-Troll WDYM by gear? My AW? It's a M15 R6 (2021), nothing happened, to me never worked in the Win11, On 10 it's okay, I upgraded to 11 and stopped work :(

**Comment by VinyLimaZ** [comment]
M15 R6 (2021)
Nothing I mean that to me never worked on Win11, I received that error message saying about the fan driver error and I'm trying to work around it, because I don't want to use the AWCC... The problem with the AWCC is that shit audio driver that bugs my mic in my stream :(

**Comment by T-Troll** [comment]
Ok, let's check.
First, try [5.5.4](https://github.com/T-Troll/alienfx-tools/releases/tag/5.5.4). In case it works, i will need your help to catch what is broken in latest.
Second, try to run `alienfan-cli rpm` from Administrator command line. Keep your eye on error message (it has many of them), it helps me to configure out what exactly happened.
In case it complains about "System configuration issue" - try to load driver manually ([as defined above](https://github.com/T-Troll/alienfx-tools/issues/114#issuecomment-1047348423) into this thread).

**Comment by VinyLimaZ** [comment]
5.5.4 didn't work, BUT I'll try to run that command to see the errors message and return to you (I'll try others versions too)

**Comment by T-Troll** [comment]
Any feedback?

**Comment by VinyLimaZ** [comment]
```
PS C:\Program Files\AlienFX Tools> .\alienfan-cli.exe
AlienFan-cli v5.6.0
System configuration issue - see readme.md for details!
```
I'll try to load manually the driver later, but the commando doesn't return anything that helps... 

The strange part is that when I reinstalled the WIN11 at first the app worked, but after the update stoped :/

**Comment by T-Troll** [comment]
Yes, it is. I'm a bit tired from this - M$ make a lot of effort to block open source drivers to load into W11.
AFAIR, situation will be even worse later.

All we can do is to try to find a way to load it anyway, it's wrong to follow their way.

So, latest KDU didn't help? Error code and driver not loaded? Or driver loaded but can't be accessed by tools?

**Comment by JessloRz** [comment]
Version 5.5.5.0 worked for me on windows 11, the only problem i had with that version is that, when i click the minimize button of the window, it crashes.
Minimizing from app button works fine

**Comment by T-Troll** [comment]
A mystery!
The only change is in alienfx-gui, but the driver, loading, and fan SDK are the same!

Maybe some Visual Studio update broke it? Hmm...
Can you try to load .zip from the latest release, then copy hwacc.sys, kdl.dll and drv64.dll from 5.5.5.0?

**Comment by JessloRz** [comment]
It works!

![Captura de pantalla 2022-04-18 200936](https://user-images.githubusercontent.com/99861741/163900512-783cd382-1234-4d1a-890c-8ab9b3aa6d29.png)

**Comment by T-Troll** [comment]
Great!
So, as a hotfix, you can always use these files from 5.5.5 (No changes inside, in fact).
I'll try to configure out what happened and maybe generate correct ones for test.

**Comment by T-Troll** [comment]
Mystery continues:

```C:\work\DevStudio\alienfx-tools\x64\Release\t>fc HwAcc.sys ../t2/HwAcc.sys
Comparing files HwAcc.sys and ../T2/HWACC.SYS
FC: no differences encountered
C:\work\DevStudio\alienfx-tools\x64\Release\t>fc kdl.dll ../t2/kdl.dll
Comparing files kdl.dll and ../T2/KDL.DLL
FC: no differences encountered
C:\work\DevStudio\alienfx-tools\x64\Release\t>fc drv64.dll ../t2/drv64.dll
Comparing files drv64.dll and ../T2/DRV64.DLL
FC: no differences encountered
```

It's between 5.5.5 and 5.5.5.1.
I have 2 ideas what happened:
1. AV broke something in installer
2. Installer broken by itself

Can you please check original 5.5.8 not from .msi, but from .zip?

**Comment by JessloRz** [comment]
Yeah, it seems like the problem is with the installer version, i didn't try downloading the .zip until now so didn't notice.

![image](https://user-images.githubusercontent.com/99861741/163912385-3f36139a-7314-4a95-a827-0745d0ef3e70.png)

**Comment by T-Troll** [comment]
Uff... This kind of issue is so hard to pinpoint.
Thank you for help!

Seems like W11 installer broke/ignore something from .msi into the process. I'll check.

**Comment by JessloRz** [comment]
> Yeah, it seems like the problem is with the installer version, i didn't try downloading the .zip until now so didn't notice.
> 
> ![image](https://user-images.githubusercontent.com/99861741/163912385-3f36139a-7314-4a95-a827-0745d0ef3e70.png)

Wait, this is actually pretty weird, tried installing again with the .msi and it works. Maybe installing 5.5.5.0 first has something to do with it? I can only help testing though haha...

**Comment by T-Troll** [comment]
A puzzle inside the mystery.
No, it's not a difference in saved data or something between it.
The only reason in this case is GitHub/Windows AV issues (broken .msi or AV block/broke something during installation).
I bet on 2nd.

Anyway, let's keep our eye, and we know the solution now.

**Comment by T-Troll** [comment]
Ohhh! WAIT!
Reboot now with 5.5.8 active. It can be a surprise!
(Driver only loaded at first app start, then stay in RAM until reboot, so if you load it once - it doesn't touch anymore).

**Comment by JessloRz** [comment]
It keeps working. I also tried uninstalling and installing a few more times but it worked every single one.
I'm out of ideas haha...

**Comment by Hellohi3654** [comment]
I think it's better if I post this here

I think a new windows update has broken the fan control again
same behavior as before

"kdu.exe -prv 20 -scv 3 -drvn HwAcc -map HwAcc.sys"
this command now works

**Comment by T-Troll** [comment]
Even more funny. I got a BSOD today at W10 - in RTCore as well. Seem like latest Windows update.
Ok, i'll switch apps to provider 20.

**Comment by T-Troll** [comment]
Uff... A bit tricky with this provider. Can you check other?

**Comment by Hellohi3654** [comment]
> Uff... A bit tricky with this provider. Can you check other?

Only provider that works for me

**Comment by T-Troll** [comment]
@Hellohi3654 - Can you please check:
- Disable CDU loading
- SHUTDOWN (not restart) system
- Start it again
Is driver loaded in this case?

**Comment by Hellohi3654** [comment]
> Uff... A bit tricky with this provider. Can you check other?

Checked the other ones almost two months later sorry about that

`kdu.exe -prv 13 -scv 3 -drvn HwAcc -map HwAcc.sys`

works along with 

`kdu.exe -prv 20 -scv 3 -drvn HwAcc -map HwAcc.sys`

Hopefully 13 easier to implenment 

> @Hellohi3654 - Can you please check:
> 
> * Disable CDU loading
> * SHUTDOWN (not restart) system
> * Start it again
>   Is driver loaded in this case?

No nothing I do makes it work still getting same error using

`kdu.exe -prv 1 -scv 3 -drvn HwAcc -map HwAcc.sys`

and just opening alienfan-gui doesn't work

**Comment by T-Troll** [comment]
Thank you. As you can see at Discord, i plan to drop driver access and use the other way.

**Comment by Hellohi3654** [comment]
> 

oh cool

**Comment by Hellohi3654** [comment]
> Thank you. As you can see at Discord, i plan to drop driver access and use the other way.

oh cool

**Comment by T-Troll** [comment]
Yes. You can check it now at #testing channel.
Seems like all works, but now not all sensors visible and G-Mode disabled.
... and no more drivers or KDL.

**Comment by Hellohi3654** [comment]
fan control works fine 
but the fan rpms are a bit off
C:\Alienfan\AlienFan_V2>alienfan-cli rpm
WMI activation started.
AlienFan-cli v6.2.1.1
AWCC section detected!
System information available!
2 Fans found
2 Temperature sensors found
2 Power modes found
Fan control available!
Supported hardware v1 detected, 2 fans, 2 sensors, 2 power states. Light control disabled.
Fan#0: 1200
Fan#1: 2700

I believe that fan#0 is doing 1200 rpm but there's no way fan#1 is doing 2700 rpm it's meant to be doing the same as fan#0 but usually fan#1 is doing like 300 rpm less than fan#0
Very weird

**Comment by Hellohi3654** [comment]
![image](https://user-images.githubusercontent.com/57390764/183285983-d8d59d00-1339-42e5-a30b-9ef050d77a71.png)

Oh also I posted in discord but I'll put here
Temp and max temp are always at the same value

**Comment by Hellohi3654** [comment]
oh and also
```
C:\Alienfan\AlienFan_V2>alienfan-cli dump
WMI activation started.
AlienFan-cli v6.2.1.1
AWCC section detected!
System information available!
2 Fans found
2 Temperature sensors found
2 Power modes found
Fan control available!
Supported hardware v1 detected, 2 fans, 2 sensors, 2 power states. Light control disabled.
Names: [WMI, dynamic: ToInstance, provider("WmiProv"), Locale("MS\\0x409"), Description("WMI Function"), guid("{A70591CE-A997-11DA-B012-B622A1EF5492}")]
class AWCCWmiMethodFunction
{
        [key, read] string InstanceName;
        [read] boolean Active;
        [WmiMethodId(13), Implemented, read, write, Description("Return Overclocking Report.")] void Return_OverclockingReport([out] uint32 argr);
        [WmiMethodId(14), Implemented, read, write, Description("Set OCUIBIOS Control.")] void Set_OCUIBIOSControl([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(15), Implemented, read, write, Description("Clear OC FailSafe Flag.")] void Clear_OCFailSafeFlag([out] uint32 argr);
        [WmiMethodId(19), Implemented, read, write, Description("Get Fan Sensors.")] void GetFanSensors([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(20), Implemented, read, write, Description("Thermal Information.")] void Thermal_Information([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(21), Implemented, read, write, Description("Thermal Control.")] void Thermal_Control([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(22), Implemented, read, write, Description("BIOS OC Control.")] void BIOSOCControl([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(23), Implemented, read, write, Description("MemoryOCControl.")] void MemoryOCControl([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(24), Implemented, read, write, Description("Set SW OC Status.")] void SetSWOCStatus([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(25), Implemented, read, write, Description("OC Default Value.")] void OCDefaultValue([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(26), Implemented, read, write, Description("System Information.")] void SystemInformation([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(28), Implemented, read, write, Description("Power Information.")] void PowerInformation([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(32), Implemented, read, write, Description("FW Update GPIO toggle.")] void FWUpdateGPIOtoggle([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(33), Implemented, read, write, Description("Read Total of GPIOs.")] void ReadTotalofGPIOs([out] uint32 argr);
        [WmiMethodId(34), Implemented, read, write, Description("Read GPIO pin Status.")] void ReadGPIOpPinStatus([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(35), Implemented, read, write, Description("Read Chassis Color.")] void ReadChassisColor([out] uint32 argr);
        [WmiMethodId(36), Implemented, read, write, Description("Read Platform Properties.")] void ReadPlatformProperties([out] uint32 argr);
        [WmiMethodId(37), Implemented, read, write, Description("Game Shift Status.")] void GameShiftStatus([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(128), Implemented, read, write, Description("Caldera SW installation.")] void CalderaSWInstallation([out] uint32 argr);
        [WmiMethodId(129), Implemented, read, write, Description("Caldera SW is released.")] void CalderaSWReleased([out] uint32 argr);
        [WmiMethodId(130), Implemented, read, write, Description("Caldera Connection Status.")] void CalderaConnectionStatus([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(131), Implemented, read, write, Description("Surprise Unplugged Flag Status.")] void SurpriseUnpluggedFlagStatus([out] uint32 argr);
        [WmiMethodId(132), Implemented, read, write, Description("Clear Surprise Unplugged Flag.")] void ClearSurpriseUnpluggedFlag([out] uint32 argr);
        [WmiMethodId(133), Implemented, read, write, Description("Cancel Undock Request.")] void CancelUndockRequest([out] uint32 argr);
        [WmiMethodId(135), Implemented, read, write, Description("Devices in Caldera.")] void DevicesInCaldera([in] uint32 arg2, [out] uint32 argr);
        [WmiMethodId(136), Implemented, read, write, Description("Notify BIOS for SW ready to disconnect Caldera.")] void NotifyBIOSForSWReadyToDisconnectCaldera([out] uint32 argr);
        [WmiMethodId(160), Implemented, read, write, Description("Tobii SW installation.")] void TobiiSWinstallation([out] uint32 argr);
        [WmiMethodId(161), Implemented, read, write, Description("Tobii SW Released.")] void TobiiSWReleased([out] uint32 argr);
        [WmiMethodId(162), Implemented, read, write, Description("Tobii Camera Power Reset.")] void TobiiCameraPowerReset([out] uint32 argr);
        [WmiMethodId(163), Implemented, read, write, Description("Tobii Camera Power On.")] void TobiiCameraPowerOn([out] uint32 argr);
        [WmiMethodId(164), Implemented, read, write, Description("Tobii Camera Power Off.")] void TobiiCameraPowerOff([out] uint32 argr);
};


C:\Alienfan\AlienFan_V2>


```

**Comment by T-Troll** [comment]
Thanks. No, maxTemps looks correct for me.
I'll check the issue about RPMs.
Thanks for dump share, it helps a lot to support other sensors correctly!

**Comment by Hellohi3654** [comment]
> fan control works fine but the fan rpms are a bit off C:\Alienfan\AlienFan_V2>alienfan-cli rpm WMI activation started. AlienFan-cli v6.2.1.1 AWCC section detected! System information available! 2 Fans found 2 Temperature sensors found 2 Power modes found Fan control available! Supported hardware v1 detected, 2 fans, 2 sensors, 2 power states. Light control disabled. Fan#0: 1200 Fan#1: 2700
> 
> I believe that fan#0 is doing 1200 rpm but there's no way fan#1 is doing 2700 rpm it's meant to be doing the same as fan#0 but usually fan#1 is doing like 300 rpm less than fan#0 Very weird

fan rpms on v1

![image](https://user-images.githubusercontent.com/57390764/183286163-0e775363-611e-4532-b024-0ffab679ec0d.png)

**Comment by T-Troll** [comment]
You can help me a lot if digging inside WMI (or CIMV) and find the point exposing KDRT sensors... I didn't find it yet.

**Comment by Hellohi3654** [comment]
> You can help me a lot if digging inside WMI (or CIMV) and find the point exposing KDRT sensors... I didn't find it yet.

what do wmi cimv and kdrt mean?

**Comment by Hellohi3654** [comment]
Also the max temp sensors display the same value as the temp value

![image](https://user-images.githubusercontent.com/57390764/183289647-57b746f5-9b82-4988-a7f3-231d7a12d7a8.png)

Basically about 10 seconds ago my CPU was at 80C but the max temp sensor is synced to the temp sensor so it just shows the current temp
It's kinda like if you just spam the clear max temp button all of time

**Comment by T-Troll** [comment]
This is mine shot for V2:
![image](https://user-images.githubusercontent.com/51940412/183289940-11fbf65f-459b-4fa8-bb42-1b50ef864deb.png)

So seems like it's you system issue. I'll check.

**Comment by T-Troll** [comment]
@Hellohi3654 - this bug is fixed into [v7.0.0](https://github.com/T-Troll/alienfx-tools/releases/tag/7.0.0)

Release v7 utilize different access method, so this issue is not relevant anymore in general.
